multiqc/modules/adapterRemoval/adapterRemoval.py

```diff
@@ -0,0 +1,311 @@
+                    pass
+            else:
+                if not self.__collapsed:
+                    self.len_dist_plot_data['mate1'][self.s_name][l_data[0]] = l_data[1]
+                    self.len_dist_plot_data['mate2'][self.s_name][l_data[0]] = l_data[2]
+                    self.len_dist_plot_data['singleton'][self.s_name][l_data[0]] = l_data[3]
+                    self.len_dist_plot_data['collapsed'][self.s_name][l_data[0]] = 0
+                    self.len_dist_plot_data['collapsed_truncated'][self.s_name][l_data[0]] = 0
+                    self.len_dist_plot_data['discarged'][self.s_name][l_data[0]] = l_data[4]
+                    self.len_dist_plot_data['all'][self.s_name][l_data[0]] = l_data[5]
+                else:
+                    self.len_dist_plot_data['mate1'][self.s_name][l_data[0]] = l_data[1]
+                    self.len_dist_plot_data['mate2'][self.s_name][l_data[0]] = l_data[2]
+                    self.len_dist_plot_data['singleton'][self.s_name][l_data[0]] = l_data[3]
+                    self.len_dist_plot_data['collapsed'][self.s_name][l_data[0]] = l_data[4]
+                    self.len_dist_plot_data['collapsed_truncated'][self.s_name][l_data[0]] = l_data[5]
+                    self.len_dist_plot_data['discarged'][self.s_name][l_data[0]] = l_data[6]
+                    self.len_dist_plot_data['all'][self.s_name][l_data[0]] = l_data[7]
+
+    def adapter_removal_stats_table(self):
+
+        headers = OrderedDict()
+        headers['percent_aligned'] = {
+                'title': '% Trimmed',
+                'description': '% trimmed reads',
+                'max': 100,
+                'min': 0,
+                'suffix': '%',
+                'scale': 'RdYlGn',
+                'format': '{:.1f}%',
+                'shared_key': 'percent_aligned',
+        }
+        headers['aligned_total'] = {
+                'title': 'Total Trimmed',
+                'description': 'total trimmed reads',
+                'min': 0,
+                'scale': 'PuBu',
+                'shared_key': 'aligned_total'
+        }
+        self.general_stats_addcols(self.adapter_removal_data, headers)
+
+    def adapter_removal_retained_chart(self):
+
+        pconfig = {
+            'title': 'retained and discarded',
+            'id': 'ar_retained_plot',
+            'ylab': '# Reads',
+            'hide_zero_cats': False,
+            'cpswitch_counts_label': 'Number of Reads'
+        }
+
+        cats_pec = OrderedDict()
+        cats_pec['singleton_m1'] = {'name': 'singleton mate1'}
+        cats_pec['singleton_m2'] = {'name': 'singleton mate2'}
+        cats_pec['retained_reads'] = {'name': 'retained read pairs'}
+        cats_pec['full-length_cp'] = {'name': 'full-length collapsed pairs'}
+        cats_pec['truncated_cp'] = {'name': 'truncated collapsed pairs'}
+        cats_pec['discarded_m1'] = {'name': 'discarded mate1'}
+        cats_pec['discarded_m2'] = {'name': 'discarded mate2'}
```

Comment:
It would be good if we could make these names shorter, as they overlap onto two lines when viewed in a not-huge window. Could perhaps use `R1` and `R2` instead of `mate1` and `mate2`? Not sure that there's much else actually..